{
  try {
    doExec(env);
  }
 catch (  EvalException|RuntimeException ex) {
    throw handleException(ex);
  }
}
